mne/viz/epochs.py

```diff
@@ -442,6 +442,194 @@ def plot_epochs(epochs, epoch_idx=None, picks=None, scalings=None,
+    total_channels = len(picks)
+    data = np.zeros((len(epochs.events), epochs.info['nchan'], len(times)))
+
+    for ii, epoch in enumerate(epoch_data):
+        j = 0
+        for jj, this_channel in enumerate(epoch):
+            if jj in picks:
+                data[ii, j] = this_channel / scalings[types[0]]
+                j += 1
+
+    # set up plotting
+    size = get_config('MNE_BROWSE_RAW_SIZE')
+    if size is not None:
+        size = size.split(',')
+        size = tuple([float(s) for s in size])
+    fig = figure_nobar(figsize=size)
+    ax = plt.subplot2grid((10, 10), (0, 0), colspan=9, rowspan=9)
+    ax.set_title(title_str, fontsize=12)
+    ax_hscroll = plt.subplot2grid((10, 10), (9, 0), colspan=9)
+    ax_hscroll.get_yaxis().set_visible(False)
+    ax_hscroll.set_xlabel('Epochs')
+    ax_vscroll = plt.subplot2grid((10, 10), (0, 9), rowspan=9)
+    ax_vscroll.set_axis_off()
+    # populate vertical and horizontal scrollbars
+    for ci in range(len(epochs.ch_names)):
+        ax_vscroll.add_patch(mpl.patches.Rectangle((0, ci), 1, 1,
+                                                   facecolor='k',
+                                                   edgecolor='k'))
+    vsel_patch = mpl.patches.Rectangle((0, 0), 1, n_channels, alpha=0.5,
+                                       edgecolor='w', facecolor='w')
+    ax_vscroll.add_patch(vsel_patch)
+    params['vsel_patch'] = vsel_patch
+
+    ax_vscroll.set_ylim(len(types), 0)
+    ax_vscroll.set_title('Ch.')
+
+    # store these so they can be fixed on resize
+    length = len(epochs.times)
+    lines = list()
+    line = [(0, 0) for x in range(length)]
+    for channel in range(n_channels):
+        lc = LineCollection(np.array([line, ] * len(epochs)), linewidths=0.5,
+                            linestyles='solid')
+        ax.add_collection(lc)
+        lines.append(lc)
+
+    event_line = ax.plot([np.nan], color='blue')[0]
+    params['event_line'] = event_line
+    params['epochs'] = epochs
+    params['lines'] = lines
+    params['n_channels'] = n_channels
+    params['n_epochs'] = n_epochs
+    params['fig'] = fig
+    params['ax'] = ax
+    params['ax_hscroll'] = ax_hscroll
+    params['ax_vscroll'] = ax_vscroll
+    params['ch_start'] = 0
+    params['t_start'] = 0
+    params['duration'] = duration
+    params['scalings'] = scalings
+    params['types'] = types
+    params['color'] = color
+    params['ch_names'] = epochs.ch_names
+    params['picks'] = picks
+    params['ch_names'] = [epochs.info['ch_names'][x] for x in picks]
+    params['bads'] = list()
+
+    # concatenation
+    epoch_data = np.concatenate(data, axis=1)
+
+    params['data'] = epoch_data
+    params['times'] = np.arange(0, len(data) * len(epochs.times), 1)
+
+    ylim = [total_channels * 2 + 1, 0]
+    # make shells for plotting traces
+    offset = ylim[0] / n_channels
+    offsets = np.arange(n_channels) * offset + 5
+    params['offsets'] = offsets
+
+    for i in range(n_epochs):
```

Comment:
I usually use at least `ei` for `epochs_index` -- it's a balance, since sometimes long names can also hurt readability